app/models/manageiq/providers/azure/cloud_manager/provision/state_machine.rb

```diff
@@ -0,0 +1,9 @@
+module ManageIQ::Providers::Azure::CloudManager::Provision::StateMachine
+  def customize_destination
```

Comment:
Do you have to override this method, because `CloudManager::Provision::StateMachine#customize_destination` calls to `associate_floating_ip`, which you dont want in your case?

If so, I wonder if the super method here might be doing more than necessary.